js/modules/k6/ws/ws.go

```diff
@@ -249,6 +246,20 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 	// Wraps a couple of channels around conn.ReadMessage
 	go socket.readPump(readDataChan, readErrChan, readCloseChan)
 
+	// we do it here as below we can panic, which translates to an exception in js code
+	defer func() {
+		socket.Close() // just in case
```

Comment:
Does this make the `conn.Close()` in the `defer` on line 224 superfluous? It's probably fine as it will be a no-op, but it would be good to have a single (or fewer) places where the connection could be closed from.

Similar to the places we call `socket.closeConnection()` from before returning, it would be great if that can be handled in the same `defer`.

Reply:
Hm ... kind of ... `handleEvent` on Line 227 can also panic :D and probably the other stuff ... maybe just this should be higher ?!? but let's do it another time - hopefully with more tests